app/src/main/java/com/example/mystore/ui/products/ProductDetailsScreen.kt

```diff
@@ -0,0 +1,207 @@
+package com.example.mystore.ui.products
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Add
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.material.icons.filled.Star
+import androidx.compose.material.icons.outlined.Cancel
+import androidx.compose.material.icons.outlined.CheckCircle
+import androidx.compose.material.icons.outlined.ShoppingCart
+import androidx.compose.material.icons.outlined.Star
+import androidx.compose.material3.Button
+import androidx.compose.material3.Card
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import coil.compose.rememberAsyncImagePainter
+import com.example.mystore.ui.theme.Green40
+import com.example.mystore.ui.theme.Purple40
+import com.example.mystore.ui.theme.Red40
+import kotlin.math.ceil
+
+
+@Composable
+fun ProductDetailsScreen() {
+    val productViewModel: ProductDetailsViewModel = hiltViewModel()
+    productViewModel.getProduct(productId = 1)
```

Comment:
Such function invocation should never happen in the middle of a composable function. You can either place it inside of an empty `LaunchedEffect` (something that's generally mostly useful if you don't have a view model):

```
LaunchedEffect(Unit) {
  productViewModel.getProduct(productId = 1)
}
```

or even better - inside the `init` of the view model as it's constructed only once.

For later homeworks: for Homework 2-2, when you'll have to navigate from the `HomeScreen` with a real `productId`, you can make use of `SavedStateHandle` in the view model to propagate the id directly to it. Example: https://stackoverflow.com/questions/67982230/jetpack-compose-pass-parameter-to-viewmodel (second answer).